src/cryptsetup/cryptsetup.c

```diff
@@ -228,6 +233,15 @@ static int parse_one_option(const char *option) {
                 if (r < 0)
                         return log_error_errno(r, "Failed to parse %s: %m", option);
 
+        } else if ((val = startswith(option, "pkcs11-uri="))) {
+
+                if (!pkcs11_uri_valid(val))
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "pkcs11-uri= parameter expects a PKCS#11 URI, refusing");
```

Comment:
Hmm, should it just ignore it? Hard failure seems harsh here.

Reply:
not sure we can continue here... if the pkcs11 stuff is defined but we can't make sense of it you cannot really unlock your device, no matter what...